source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java

```diff
@@ -57,11 +63,10 @@ class NPCFactoryTest {
             "images/snake.atlas",
             "images/eagle.atlas",
             "images/turtle.atlas",
+            "images/Fish.atlas",
             "images/final_boss_kangaroo.atlas"
     };
 
-
-
     @BeforeEach
     void setup() {
```

Comment:
This before each shouldn't really be a before each, as you are creating every entity repeatedly before every test (this test file takes forever because of this). I would suggest creating the entities outside of the before each, and only creating new entities when needed (which I don't think any tests need).

Reply:
Thank you! This has been resolved to a BeforeAll which runs once!